src/hackney_http.erl

```diff
@@ -325,7 +328,40 @@ parse_trailers(St, Acc) ->
     _ -> error
   end.
 
-parse_body(St=#hparser{body_state=waiting, te=TE, clen=Length,
+-define(MAX_WBITS, 15).	% zconf.h
+parse_body(St=#hparser{body_state=waiting, ce=CE, clen=Length, method=Method}) ->
```

Comment:
I think we have some inconsistency here: body will be unzipped regardless of `compress` option. But maybe we should check that both `Content-Encoding` header and `compress` option are active to decompress the body?
Because, with your current implementation, there will be no way at all to get compressed body as is - without decompression. For example if I need to get compressed body for some reason (eg, send it further or store on disk without looking into it... proxy...) I may set `Accept-Encoding` manualy and will get compressed body in response. But if I need transparent compression, I'll use `compress` option.

Reply:
Good catch, I'll get back with a fix in a day or two.

Reply:
@jimdigriz did you fix it?

Reply:
I took a stab at it but noticed that to push the 'do transparent compression' flag into `#hparser{}` is quite invasive.  I do plan to return to this but real life has caught up with me. :)

Reply:
hrm which also let me think i should revisit that data structure, it's quite duplicating the `#client{}` one ...

Reply:
I really appreciate the work you are doing, this is something that I am needing. A couple questions for you.
So the `compress` option  means both to compress the request _and_ to decompress the response? I see these as two different concerns. Maybe I've missed something reading through the comments.
I agree with seriyps above, I would only want to decompress the response if I pass in the option.
One one case that I have, I don't really care about my request being compressed, but I care about the response being uncompressed. When I see the option `compress` I think there might be another option for the response. Thinking out loud.
Appreciate the work you have done, I'm wanting to get this into prod soon myself.

Reply:
I agree completely too with @seriyps thoughts too, it was something I had not thought of at the time.  As @benoitc noticed too, to resolve this will probably involve a substantial change and I have not really looked much into it since.

@benoitc my concern is only wasting everyones time.  If you can describe the changes you would like to see to replumb this, when I find the time to progress this, I would then be able to move in a direction that works for you.

Reply:
@benoitc Any idea how to help @jimdigriz ^?

Reply:
@jimdigriz I believe this was resolved with the last commit? Unfortunately, I can't see what changed in the last commit because of force-push..